### PR TITLE
Get actual chain config from db data

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
If chain is already (partially) sync'ed we should get `ChainConfig` from database.

### CODE REVIEW

File: db/silkworm/db/access_layer.cpp [hunk 1]
```diff
@@ -355,4 +356,77 @@ bool migration_happened(lmdb::Transaction& txn, const char* name) {
...
+    */
+
+    auto config_json{json::parse(config_value->c_str())};
+    ChainConfig config{};
+
+    if (config_json.contains("chainId")) {
+        config.chain_id = config_json["chainId"].get<uint64_t>();
+    }
+    if (config_json.contains("homesteadBlock")) {
+        config.homestead_block.emplace(config_json["homesteadBlock"].get<uint64_t>());
```
Comment (yperbasis) [inline db/silkworm/db/access_layer.cpp:399]: I'd simply say
```
config.homestead_block = config_json["homesteadBlock"].get<uint64_t>();
```
but that's a matter of taste.

**Review by yperbasis — APPROVED** [review]

File: db/silkworm/db/access_layer.cpp (hunk 1 above)
Comment (AndreaLanfranchi) [reply db/silkworm/db/access_layer.cpp:399]: It throws if member does not exists

Comment (AndreaLanfranchi) [reply db/silkworm/db/access_layer.cpp:399]: Sorry I misread the comment ... yes maybe is a matter of taste. But as far as I understand emplace does not create a new std::optional and copies it to the member.

Comment (yperbasis) [reply db/silkworm/db/access_layer.cpp:399]: I don't understand. How
```
config.homestead_block = config_json["homesteadBlock"].get<uint64_t>();
```
is different it terms of throwing from
```
config.homestead_block.emplace(config_json["homesteadBlock"].get<uint64_t>());
```
?

Comment (AndreaLanfranchi) [reply db/silkworm/db/access_layer.cpp:399]: Read my above. Sorry I posted it late

Comment (yperbasis) [reply db/silkworm/db/access_layer.cpp:399]: No worries, both `emplace` and `=` are fine. I suspect the optimizer will produce the same code, but it's a minor thing.

**Pull request merged**